### PR TITLE
Npm code refactor

### PR DESCRIPTION
**Pull request by sarao1310**
- [x] All [tests](https://github.com/jfrog/jfrog-cli#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I'll happily review this PR once the tests are passing.

File: buildtools/cli.go [hunk 1]
```diff
@@ -822,23 +801,42 @@ func NpmInstallCmd(c *cli.Context) error {
...
 func NpmCiCmd(c *cli.Context) error {
 	if show, err := cliutils.ShowGenericCmdHelpIfNeeded(c, c.Args(), "npmcihelp"); show || err != nil {
 		return err
 	}
-	return npmInstallCiCmd(c, npm.NewNpmCiCommand())
+	return npmGenericCmd(c, "ci", true)
 }
 
-func npmInstallCiCmd(c *cli.Context, npmCmd *npm.NpmInstallOrCiCommand) error {
+func npmGenericCmd(c *cli.Context, cmdName string, collectBuildInfo bool) error {
```
Comment (eyalbe4) [inline buildtools/cli.go:814]: collectBuildInfo --> collectBuildInfoIfRequested

File: buildtools/cli.go [hunk 2]
```diff
@@ -822,23 +801,42 @@ func NpmInstallCmd(c *cli.Context) error {
...
 		return err
 	}
-
-	npmCmd.SetConfigFilePath(configFilePath).SetArgs(args)
+	npmCmd.SetConfigFilePath(configFilePath)
+	npmCmd.CommonArgs.SetNpmArgs(orgArgs)
+	npmCmd.CommonArgs.SetNpmArgs(args)
 	err = npmCmd.Init()
 	if err != nil {
 		return err
```
Comment (eyalbe4) [inline buildtools/cli.go:842]: Can be simplified to - 
```
	if err = npmCmd.Init(); err != nil {
		return err
	}
```
The same goes for the code in the `NpmPublishCmd` function.

File: buildtools/cli.go [hunk 3]
```diff
@@ -822,23 +801,42 @@ func NpmInstallCmd(c *cli.Context) error {
...
+	npmCmd := npm.NewNpmCommand(cmdName, collectBuildInfo)
 	configFilePath, args, err := GetNpmConfigAndArgs(c)
 	if err != nil {
 		return err
 	}
-
-	npmCmd.SetConfigFilePath(configFilePath).SetArgs(args)
+	npmCmd.SetConfigFilePath(configFilePath)
+	npmCmd.CommonArgs.SetNpmArgs(orgArgs)
+	npmCmd.CommonArgs.SetNpmArgs(args)
```
Comment (eyalbe4) [inline buildtools/cli.go:839]: The above libes can be simplified to - 
```
npmCmd.SetConfigFilePath(configFilePath).CommonArgs.SetNpmArgs(orgArgs).SetNpmArgs(args)
```

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: buildtools/cli.go (hunk 2 above)
Comment (sarao1310) [reply buildtools/cli.go:842]: Done

File: buildtools/cli.go (hunk 1 above)
Comment (sarao1310) [reply buildtools/cli.go]: Done

**Pull request merged**